src/benchmark_api_internal.cc

```diff
@@ -16,7 +16,7 @@ BenchmarkInstance::BenchmarkInstance(Benchmark* benchmark, int family_idx,
       per_family_instance_index_(per_family_instance_idx),
       aggregation_report_mode_(benchmark_.aggregation_report_mode_),
       args_(args),
-      time_unit_(benchmark_.time_unit_),
+      time_unit_(benchmark_.GetTimeUnit()),
```

Comment:
Another possible weak point is the availability of both `time_unit_` and `GetTimeUnit()`. `BenchmarkInstance` is a friend of `Benchmark`, thus it can access both, and it's unclear which one to use.

Reply:
Using friend is already a weak point here so I think it is not our job to fix that.

Reply:
I don't think so. Friends have their place, and can be useful in certain situation. In this case the problem is not with the friend, but with the introduction of a separate getter method.

Reply:
I meant in this case. Since these Benchmark objects are globals they are initialized before the main function. I couldn't think of an alternative solution that wasn't much more complex.